packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteDatabase.ts

```diff
@@ -0,0 +1,283 @@
+import * as SQLite from 'expo-sqlite';
+import { ConsoleLogger as Logger } from '@aws-amplify/core';
+import { PersistentModel } from '@aws-amplify/datastore';
+import { ParameterizedStatement } from '../common/SQLiteUtils';
+import { CommonSQLiteDatabase } from '../common/types';
+
+const logger = new Logger('ExpoSQLiteDatabase');
+
+const DB_NAME = 'AmplifyDatastore';
+const DB_DISPLAYNAME = 'AWS Amplify DataStore SQLite Database';
+
+// TODO: make these configurable
+const DB_SIZE = 200000;
+const DB_VERSION = '1.0';
+
+/*
+
+Note: 
+I purposely used arrow functions () => {} in this class as expo-sqlite library is not promisified
+
+*/
+
+class ExpoSQLiteDatabase implements CommonSQLiteDatabase {
+	private db: SQLite.WebSQLDatabase;
+
+	public async init(): Promise<void> {
+		// only open database once.
+		if (!this.db) {
+			this.db = SQLite.openDatabase(
+				DB_NAME,
+				DB_VERSION,
+				DB_DISPLAYNAME,
+				DB_SIZE
+			);
+		}
+	}
+
+	public async createSchema(statements: string[]): Promise<void> {
+		return await this.executeStatements(statements);
+	}
+
+	public async clear(): Promise<void> {
+		logger.debug('Clearing database');
+		// delete database is not supported by expo-sqlite. alternative way is to get all table names and drop them.
+		this.dropAllTables();
+		logger.debug('Database cleared');
+		// closing db is not required as we are not deleting the db.
+	}
+
+	public async get<T extends PersistentModel>(
+		statement: string,
+		params: any[]
+	): Promise<T> {
+		const resultSet: SQLite.SQLResultSet = await new Promise(
+			(resolve, reject) => {
+				this.db.readTransaction(tx => {
+					tx.executeSql(
+						statement,
+						params,
+						(_, result) => resolve(result),
+						(_, err) => {
+							reject(err);
+							return true;
+						}
+					);
+				});
+			}
+		);
+		const result =
+			resultSet &&
+			resultSet.rows &&
+			resultSet.rows.length &&
+			resultSet.rows._array;
+
+		return result[0] || undefined;
+	}
+
+	public async getAll<T extends PersistentModel>(
+		statement: string,
+		params: any[]
+	): Promise<T[]> {
+		const resultSet: SQLite.SQLResultSet = await new Promise(
+			(resolve, reject) => {
+				this.db.readTransaction(tx => {
+					tx.executeSql(
+						statement,
+						params,
+						(_, result) => {
+							resolve(result);
+						},
+						(_, err) => {
+							reject(err);
+							return true;
+						}
+					);
+				});
+			}
+		);
+		const result =
+			resultSet &&
+			resultSet.rows &&
+			resultSet.rows.length &&
+			resultSet.rows._array;
+		return result || [];
+	}
+
+	public async save(statement: string, params: any[]): Promise<void> {
+		await new Promise((resolve, reject) => {
```

Comment:
I believe you should be returning this `Promise` instead
```suggestion
		return new Promise((resolve, reject) => {
```

Reply:
Yes. I will update that.